src/main/java/spoon/support/reflect/declaration/CtRecordImpl.java

```diff
@@ -92,6 +100,51 @@ && getField(component.getSimpleName()).isImplicit()) {
 		return this;
 	}
 
+	@Override
+	public CtRecord canonicalConstructor() {
+
+		CtTypeReference<?>[] typeReferences =
+			getFields().stream()
+				.filter(CtElement::isImplicit)
+				.map(CtTypedElement::getType)
+				.toList().toArray(new CtTypeReference[0]);
+
+		// If a canonical constructor is not already present, adds one
+		CtConstructor<?> constructor = getConstructor(typeReferences);
+		if (constructor == null) {
+			CtConstructor<?> canonical = getFactory().createConstructor();
+			canonical.setImplicit(true);
+			canonical.setExtendedModifiers(Collections.singleton(new CtExtendedModifier(ModifierKind.PUBLIC, true)));
```

Comment:
`List.of` and  `CtExtendedModifier#implict`

Reply:
Thanks for the review. Commit done.